### ISSUE TITLE
Language server treats transient invalid didChange content as a JSON-RPC handler error

### ISSUE THREAD

**Issue by chernetskyi** [terraform-linters/tflint#2640]
### Summary

When a document is temporarily invalid during normal typing, `tflint --langserver` returns an error from the `textDocument/didChange` notification handler. The JSON-RPC layer consequently logs it as a notification handling error.

Transient syntax errors and incomplete references are unavoidable while editing. The language server should treat them as normal document states—either publishing parser diagnostics or skipping linting until the document is valid—rather than reporting an internal handler failure for every keystroke.

Steps:

1. Start TFLint as a language server.
2. Open the valid configuration below.
3. Temporarily remove the value after `example =`, as commonly happens while editing.
4. Inspect stderr.

Expected: no JSON-RPC handler error is logged for the transient invalid state.

Actual: every `didChange` received while the document is invalid produces a handler error.

Related to the general LSP umbrella issue #1356, but that issue does not cover transient edit handling.

### Command

tflint --langserver

### Terraform Configuration

```terraform
locals {
  example = "value"
}
```

### TFLint Configuration

```HCL
TFLint defaults
```

### Output

```Shell
jsonrpc2 handler: notification "textDocument/didChange" handling error: Failed to load configurations: main.tf:2,12-3,1: Missing expression; Expected the start of an expression, but found an invalid expression token.


Incomplete references similarly produce errors such as:


jsonrpc2 handler: notification "textDocument/didChange" handling error: Failed to prepare rule checking: main.tf:2,13-19: Reference to undeclared local value
```

### TFLint Version

0.61.0

### Terraform Version

1.16.0

### Operating System

- [ ] Linux
- [x] macOS
- [ ] Windows
